serverlessworkflow/modules/ROOT/pages/persistence/persistence-with-postgresql.adoc

```diff
@@ -1 +1,166 @@
+// Metadata:
+:description: Running Serverless Workflow service using PostgresSQL
+:keywords: kogito, workflow, quarkus, serverless, quarkus-cli, persistence, postgresql
+// Referenced documentation pages.
+:getting_started_create_first_workflow_guide: xref:getting-started/create-your-first-workflow-service.adoc
+:persistence_with_postgresql_guide: xref:persistence/persistence-with-postgresql.adoc
+// External pages
+:quarkus_datasource_guide: https://quarkus.io/guides/datasource
+:quarkus_datasource_devservices_guide: https://quarkus.io/guides/databases-dev-services
+:kogito_sw_callback_example: {kogito_sw_examples_url}/serverless-workflow-callback-quarkus
+:postgresql_url: https://www.postgresql.org/
+:postgresql_doc_url: https://www.postgresql.org/docs/current/
+
+This document describes how to run your Kogito Serverless Workflow application using PostgreSQL persistence.
+
+.Prerequisites
+* A Kogito Serverless Workflow service. More information available in {getting_started_create_first_workflow_guide}[Creating your first Serverless Workflow service] guide.
+* Docker is installed.
+* link:{postgresql_doc_url}[PostgreSQL] is installed. For information about PostgreSQL installation and configuration, see https://www.postgresql.org/docs/current/[PostgreSQL documentation].
+
+[NOTE]
+====
+Although PostgreSQL installation is mentioned as a requirement, this guide relies on running it as a Docker service instead.
+====
+
+Running a Kogito Serverless Workflow application with persistence enabled, is recommended approach for when your workflow execution requires wait states. For example, when a process reaches a `callback` or needs to wait for an `event`, the execution is paused and the Kogito engine takes a complete snapshot of the workflow data. This snapshot is persisted in the database as a binary format along with some metadata information about the process. This metadata includes information such as process ID, process instance ID, and process version.
+
+Runtime persistence is intended primarily for storing data that is required to resume workflow execution for a particular process instance. Once a process completes, its data is removed from the database. This persistence behavior means that only the information that is required to resume execution is persisted.
+
+In Kogito, persistence is enabled through addons. Although multiple addons are available for persistence, this guide focus on using the `kogito-addons-quarkus-persistence-jdbc` addon based on JDBC along with PostgreSQL. This addon, also extends on the Quarkus capabilities for JDBC and you can also use any of the features available directly from Quarkus JDBC support. For more information on Quarkus and JDBC, visit {quarkus_datasource_guide}[Quarkus Datasources].
+
+[[postgresql-persistence-example]]
+== Solution
+We recommend that you follow the instructions in the next sections and create applications step by step. However, you can go right to the completed example.
```

Comment:
We can suggest the user proceed with these steps without relying on cloning the example project. We add the example usually only for reference. The user would ideally follow the guide to enhance their project.

I think you are already doing this here, but can we make it more transparent?

Reply:
@ricardozanini I think we need a consistent way to do this across guides, my goal here was actually to copy and paste what Quarkus has and use that as a starting point so we could use it in all guides. wdyt?

Reply:
It's fine to have what Quarkus has, but the problem was that I had to read a few times to understand that I could use the example as the solution for the procedure.

Reply:
@ricardozanini have a look if better now